src/components/app.tsx

```diff
@@ -122,6 +136,31 @@ export class App extends React.PureComponent<IProps, IState> {
           activityName={activity.name}
           singlePage={activity.layout === ActivityLayouts.SinglePage}
         />
+        { authError === ""
+          ? this.renderNavAndContent(activity, currentPage, totalPreviousQuestions, fullWidth)
+          : <AuthError />
```

Comment:
Purely a matter of taste, but I would probably write this as:
```typescript
  { authError
      ? <AuthError />
      : this.renderActivityContent(activity, currentPage, totalPreviousQuestions, fullWidth)
  }
```
Rationale:
1. All else being equal I think conditions read better in the affirmative than the negative, i.e. "If there's an error..." rather than "If there's no error/the error is empty...".
2. I share your hesitation about `renderNavAndContent()` as a name and it doesn't seem completely unreasonable to think of the navigation as part of the `ActivityContent`.

At the end of the day these are minor nits about which different opinions are reasonable, so do what makes sense to you.

Reply:
I see, I think that makes sense about using affirmative over negative and I've made that change. Thanks.

I misread the part about changing the function name though before making the commit. I thin that makes sense too and I'll make another commit for that.